src/composables/Project.js

```diff
@@ -725,31 +736,49 @@ export async function gitCommit(projectId, message) {
  */
 export async function gitGlobalUpload(project) {
   const nowDate = new Date();
-  const currentBranch = await getCurrentBranch(project.id);
   const newBranch = `leto-modelizer_${nowDate.getTime()}`;
+  const files = (await getStatus(project.id));
 
-  await createBranchFrom(
-    project.id,
-    newBranch,
-    currentBranch,
-  );
-
-  await gitCheckout(project.id, newBranch);
-
-  const files = (await getStatus(project.id))
+  const modifiedFiles = files
     .filter((file) => file.hasUnstagedChanged
       || file.isUntracked
       || file.isUnstaged
       || file.isStaged)
     .map((file) => file.path);
 
-  await gitAdd(project.id, files);
-
-  await gitCommit(
+  await gitAdd(project.id, modifiedFiles);
+
+  /* Special case for deleted files.
+  Unlike usual git, we CAN NOT add (git add) a deleted file,
+  so here, the idea is to use the remove method of isomorphic-git to stage the
+  deleted files.
+  cf: https://github.com/isomorphic-git/isomorphic-git/issues/1099
+  */
+  const deletedfiles = files.filter((file) => file.isDeleted).map((file) => file.path);
+  await gitRemove(project.id, deletedfiles);
+
+  /*
+  Commiting BEFORE creating a new branch.
+  Again, due to some specific behavior in isomorphic-git, the modifications are not
+  transfered (unlike usual git) while creating a new branch or doing a checkout.
+
+  Creating the (orphan) commit before, give the possibility to create a new branch from that commit.
+  Which will contains all modifications AND will be attached to the newly created branch.
+  */
+  const sha1 = await gitCommit(
     project.id,
     `leto-modelizer ${nowDate.toDateString()}`,
+    true,
+  );
+
+  await createBranchFrom(
+    project.id,
+    newBranch,
+    sha1,
   );
 
+  await gitCheckout(project.id, newBranch);
+
```

Comment:
delete empty line

Reply:
There is the gitpush after the emtpy line, should I still remove it ?

Reply:
JUST DO IT !